clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -364,7 +364,7 @@ private boolean maybeSendTransactionalRequest(long now) {
                 }
             } catch (IOException e) {
                 log.debug("Disconnect from {} while trying to send request {}. Going " +
-                        "to back off and retry", targetNode, requestBuilder);
+                        "to back off and retry. Encountered {}", targetNode, requestBuilder, e);
```

Comment:
From my understanding, this must either be without the last `{}` or using `e.toString()`?  (Cf. https://stackoverflow.com/questions/6371638/slf4j-how-to-log-formatted-message-object-array-exception)